math/src/polynomial.rs

```diff
@@ -557,8 +591,31 @@ impl<F: IsField> ops::Sub<&Polynomial<FieldElement<F>>> for FieldElement<F> {
     }
 }
 
+// TODO: move to another place
+// Source: https://en.wikipedia.org/wiki/Modular_multiplicative_inverse#Multiple_inverses
+fn inplace_batch_inverse<F: IsField>(numbers: &mut [FieldElement<F>]) {
+    if numbers.is_empty() {
+        return;
+    }
+    let count = numbers.len();
+    let mut prod_prefix = Vec::with_capacity(count);
+    prod_prefix.push(numbers[0].clone());
+    for i in 1..count {
+        prod_prefix.push(&prod_prefix[i - 1] * &numbers[i]);
+    }
+    let mut bi_inv = prod_prefix[count - 1].inv();
+    for i in (1..count).rev() {
+        let ai_inv = &bi_inv * &prod_prefix[i - 1];
+        bi_inv = &bi_inv * &numbers[i];
+        numbers[i] = ai_inv;
+    }
+    numbers[0] = bi_inv;
+}
+
```

Comment:
Possibly move this to field. It will be useful in other places as well.

Reply:
Moved it to a new module `lambdaworks_math::batch_inverse` (similar to `msm`)